src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioSolutionSizeTracker.cs

```diff
@@ -4,24 +4,27 @@
 using System.Composition;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.CodeAnalysis.SolutionSize;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.SolutionSize
+namespace Microsoft.VisualStudio.LanguageServices
 {
     /// <summary>
     /// Track approximate solution size.
     /// </summary>
-    [Export(typeof(ISolutionSizeTracker))]
-    [ExportIncrementalAnalyzerProvider(nameof(SolutionSizeTracker), new[] { WorkspaceKind.Host }), Shared]
-    internal class SolutionSizeTracker : IIncrementalAnalyzerProvider, ISolutionSizeTracker
+    [ExportWorkspaceService(typeof(ISolutionSizeTracker), ServiceLayer.Host)]
+    [ExportIncrementalAnalyzerProvider(nameof(VisualStudioSolutionSizeTracker), new[] { WorkspaceKind.Host }), Shared]
+    internal class VisualStudioSolutionSizeTracker : IIncrementalAnalyzerProvider, ISolutionSizeTracker
```

Comment:
Why this change and move to VS layer? Other hosts could use this just fine in the past, like VSMac.

Reply:
How did that work @Therzok ?  The impl that returned 'true' did a test if it was in a VSWorkspace.

Reply:
I'm not sure what you mean. The old code would query `IPersistentStorageLocationService` which handled getting the database location.

Now, we're stuck with having to source copy the VS implementation instead of properly providing a host-specific implementation for that.

Reply:
Can you point me at the code you are talking about in VS4Mac?  I'm not really understanding what you are saying.  All that moved to the vs layer was the size-tracker.  that was only used by the persistence service, and only for VS since thata was the only host that returned `true` for IsSupported.

Reply:
Before this change, just defining an `IPersistentStorageLocationService` implementation would yield us with a functioning `SolutionSizeTracker`. VSMac uses the persistence layer and the sqlite implementation that is also used in Windows.

Without a solution size tracker implementation, this means that [persistence is disabled](https://github.com/dotnet/roslyn/blob/1de29ff0ae5661a0ac4b8208ac42570bce4c9417/src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs#L151-L153).

> and only for VS since thata was the only host that returned true for IsSupported.

Yes, VS `PersistentStorageLocationService` would set `IsSupported` for `VisualStudioWorkspace` due to the composition including this host component.  But we do it in [VSMac](https://github.com/mono/monodevelop/blob/c740bfab1077385701ebf3486970fef4d9e44a86/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopPersistentStorageLocationService.cs#L89) too, enabling it for our workspace type.

Having to source copy code like the `SolutionSizeTracker` implementation can end up with unsynced implementations between the IDEs.